Sally7.Benchmarks/Serialization/SerializePrimitivesAsLargerPrimitive.cs

```diff
@@ -15,9 +15,9 @@ public void SerializeOneByOne()
     {
         ref var destination = ref MemoryMarshal.GetReference(buffer.AsSpan());
         uint offset = 0;
-        for (var i = 0; i < value.Length; i++)
+        foreach (var val in value)
```

Comment:
Ah, the reason is that `value` is a field and not a local. So with the `foreach`-loop the C# compiler pulls the field into a local thus better code can be generated.

BTW: that's one of the reasons why I prefer the `_` prefix for fields, to easily distinguish between fields and locals. The in the previous code (with the `for`-loop) it would be obvious (for me) 😉.

Reply:
Yeah I need to start using `_` prefix. Years ago my manager decided we shouldn't use underscore prefixes, I've been regretting that decision for a while now. I stuck to that convention for my private projects as well, but I totally agree with what you're saying.

What I though was kinda funny was dat for the `...Combined` method the `foreach` didn't make any difference at all, the JIT is still generating the same instructions.

Reply:
I think the JIT is quite sensitive here, or in other words it's luck when the JIT generates the same code here.
Some reasons for this are that the JIT can prove the field-access is safe, null-checks, etc. Best and most readable option here (array field) is the `foreach`-loop.